dns/message.py

```diff
@@ -696,7 +711,96 @@ def _parse_special_rr_header(self, section, count, position,
 
 
 class QueryMessage(Message):
-    pass
+    def resolve_chaining(self):
+        """Follow the CNAME chain in the response to determine the answer
+        RRset.
+
+        Raises NotQueryResponse if the message is not a response.
+
+        Raises dns.message.ChainTooLong if the CNAME chain is too long.
```

Comment:
Any reason that `ChainTooLong` is fully qualified, and the other exceptions aren't?